lib/spack/spack/installer.py

```diff
@@ -1716,6 +1716,16 @@ def install(self):
                 spack.hooks.on_install_cancel(task.request.pkg.spec)
                 raise
 
+            except binary_distribution.NoChecksumException as exc:
+                if not task.request.install_args.get('cache_only'):
+                    # Checking hash on downloaded binary failed.
+                    err = 'Failed to install {0} due to {1}:'
+                    err += ' Requeue for manual installation.'
```

Comment:
Would something along the lines of the following be more accurate?

```suggestion
                    err = 'Failed to install {0} from binary cache due to {1}:'
                    err += ' Requeueing to install from source.'
```